service/controller/chartconfig/v7/resource/chartstatus/create.go

```diff
@@ -29,30 +29,41 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	if key.ReleaseStatus(customObject) != releaseContent.Status {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, releaseContent.Status))
-
-		customObjectCopy := customObject.DeepCopy()
-		customObjectCopy.Status.ReleaseStatus = releaseContent.Status
+	var status, reason string
+	{
+		if key.IsCordoned(customObject) {
+			status = releaseStatusCordoned
+			reason = key.CordonReason(customObject)
+		} else {
+			status = releaseContent.Status
+			if releaseContent.Status != releaseStatusDeployed {
+				releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
+				if err != nil {
+					return microerror.Mask(err)
+				}
 
-		if releaseContent.Status != releaseStatusDeployed {
-			releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
-			if err != nil {
-				return microerror.Mask(err)
+				reason = releaseHistory.Description
+			} else {
+				reason = ""
 			}
-
-			customObjectCopy.Status.Reason = releaseHistory.Description
-		} else {
-			customObjectCopy.Status.Reason = ""
 		}
+	}
+
+	if key.ReleaseStatus(customObject) != status {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, status))
+
+		customObjectCopy := customObject.DeepCopy()
+		customObjectCopy.Status.ReleaseStatus = status
+		customObjectCopy.Status.Reason = reason
+
 		_, err := r.g8sClient.CoreV1alpha1().ChartConfigs(customObject.Namespace).UpdateStatus(customObjectCopy)
```

Comment:
So far we always fetch the CR again to have the latest resource version. When the status update now fails because someone updated the CR meanwhile, then the retry resource will execute this here again, which will still fail because the resource version is still the same outdated one.

Reply:
I would do the same practice for other operators to prevent that situation, thanks to catching it!